src/utils/useUserData.js

```diff
@@ -30,7 +30,7 @@ const useUserData = () => {
     const [loading, setLoading] = useState(true);
 
     useEffect(() => {
-        const unsubscribe = onAuthStateChanged(auth, async (currentUser) => {
+        const unsubscribe = onAuthStateChanged(auth, async (currentUser) => { 
```

Comment:
As we discussed, you should separate the user data to the authentication aspect.